apps/src/assetManagement/IconListEntry.jsx

```diff
@@ -0,0 +1,58 @@
+var Icon = require('./Icon.jsx');
+
+/**
+ * A list of icons, maybe filtered by a search query.
+ */
+module.exports = React.createClass({
+  propTypes: {
+    assetChosen: React.PropTypes.func.isRequired,
+    iconId: React.PropTypes.string.isRequired,
+    altMatch: React.PropTypes.string.isRequired,
+    search: React.PropTypes.string.isRequired
+  },
+
+  highlightSearch: function (str) {
+    var offset = str.indexOf(this.props.search);
+    if (offset === -1) {
+      return str;
+    }
+    var left = str.substr(0, offset), right = str.substr(offset + this.props.search.length);
+    return [left, <span style={{backgroundColor: '#ffc'}}>{this.props.search}</span>, right];
+  },
+
+  render: function () {
+
+    var iconLabel, columnWidth;
+
+    if (this.props.search) {
+      var query = new RegExp(this.props.search), altMatchText;
+      if (!query.test(this.props.iconId)) {
+        // We matched based on an alternate keyword, show that keyword next to the icon ID.
+        altMatchText = <p style={{
+          float: 'left',
+          fontSize: '13px',
+          color: '#999'
+        }}>({this.highlightSearch(this.props.altMatch)})</p>;
+      }
+
+      iconLabel = [
+        <p style={{float: 'left', margin: '0 5px', fontSize: '13px', color: '#000'}}>{this.highlightSearch(this.props.iconId)}</p>,
+        altMatchText
+      ];
+
+      columnWidth = '33%';
```

Comment:
Default `columnWidth` is null, but in search result mode there should be 3 equal columns.